src/mapInteractor.js

```diff
@@ -850,18 +850,27 @@ var mapInteractor = function (args) {
         m_mouse.buttons[prop] = false;
       }
     }
-    if (evt.type !== 'mouseup') {
+    /* If the event buttons are specified, use them in preference to the
+     * evt.which for determining which buttons are down.  buttons is a bitfield
+     * and therefore can represent more than one button at a time. */
+    if (evt.buttons !== undefined) {
+      m_mouse.buttons.left = !!(evt.buttons & 1);
+      m_mouse.buttons.right = !!(evt.buttons & 2);
+      m_mouse.buttons.middle = !!(evt.buttons & 4);
+    } else if (evt.type !== 'mouseup') {
+      /* If we don't evt.buttons, fall back to which, but not on mouseup. */
       switch (evt.which) {
         case 1: m_mouse.buttons.left = true; break;
         case 2: m_mouse.buttons.middle = true; break;
         case 3: m_mouse.buttons.right = true; break;
-        default:
-          if (evt.which) {
-            m_mouse.buttons[evt.which] = true;
-          }
-          break;
       }
     }
+    /* When handling touch events, evt.which can be a string, in which case
+     * handle a "button" with that name -- a non-integer string will not
+     * evaluate as being between 1 and 3. */
+    if (evt.which && !(evt.which >= 1 && evt.which <= 3)) {
```

Comment:
Should this be a separate conditional block, whereby one of the other previous conditionals runs with this too?

Reply:
It doesn't need to be, as in the touch events where it will get triggered, it can never also be in a `mouseup` event, and we want it to trigger regardless of the buttons conditional.